sdk/servicebus/azure-servicebus/azure/servicebus/common/mixins.py

```diff
@@ -312,11 +322,11 @@ def from_connection_string(cls, conn_str, name=None, **kwargs):
         :param name: The name of the entity, if the 'EntityName' property is
          not included in the connection string.
         """
-        address, policy, key, entity = parse_conn_str(conn_str)
+        address, policy, key, entity, transport_type = parse_conn_str(conn_str)
         entity = name or entity
         address = build_uri(address, entity)
         name = address.split('/')[-1]
-        return cls(address, name, shared_access_key_name=policy, shared_access_key_value=key, **kwargs)
+        return cls(address, name, shared_access_key_name=policy, shared_access_key_value=key, transport_type=transport_type, **kwargs)
```

Comment:
If it wouldn't be a highly vacuous/infeasible test to write, would be nice to have a quick validation that passing the various transport_types works as expected.

Reply:
To be completely honest here, I'm not sure how to go about this. Should I just test the `auth_config` property to see if it got the transport_type key set correctly? If it needs to actually connect to a test service bus, I'm not familiar with how to wire up the test for that.

Reply:
Would this suffice?

```python
def test_servicebusclient_from_conn_str_amqpoverwebsocket():
    conn_str = f'Endpoint={test_addr}/;SharedAccessKeyName={test_key_name};SharedAccessKey={test_key};TransportType=AmqpOverWebsocket'
    sb_client = ServiceBusClient.from_connection_string(conn_str)
    assert sb_client.transport_type == TransportType.AmqpOverWebsocket
```

Reply:
So I'll make no pretenses that getting the test framework set up to run live tests is clear; I've had on my long-term TODO list surfacing some docs for that for some time now.

What you described would be fine but you'd need to leverage the Preparers; I'm going to be running some integration tests on your PR before hitting approve (As a standard sanity check) so assuming you allowed maintainers to edit your branch, I can throw together a similar snippet and use it as an opportunity for you to see the structure in action so you'll be appraised for the future.  (I figure it's the least I can do/will save us a bunch of back and forth :)  )

Reply:
Thank you, that sounds great. Let me check to ensure maintainers can edit my branch now.